scripts/home.js

```diff
@@ -1,34 +1,545 @@
+    searchBox.classList.remove("hide-search-box");
+    searchInputField.value = "";
+  } else {
+    searchBox.classList.add("hide-search-box");
+  }
+});
+
+// Remove the menu and search box
+// if screen width > 550
+window.onresize = () => {
+  if (window.innerWidth > 550) {
+    if (menuContent.classList.contains("show")) {
+      menuContent.classList.add("hide");
+      menuContent.classList.remove("show");
     }
-});  
+    if (burgerMenu.classList.contains("open")) {
+      burgerMenu.classList.remove("open");
+    }
+    searchBox.classList.add("hide-search-box");
+  }
+};
 
-// prevent page from reload when the search form get submitted 
-document.querySelector("#header .search-bar form").addEventListener("submit",event=>{
-    event.preventDefault(); 
-    searchInputField.value = "";
+// Filter By Category
+// Add event listener to the utilites box (big and medium screens)
+document
+  .querySelector(".utilities-box__content .categories")
+  .addEventListener("click", (event) => {
+    const categoryTitle = event.target.title;
+    if (categoryTitle) {
+      productsCurrentlyShown = filterByCategory(categoryTitle);
+      populateProduct(productsCurrentlyShown);
+    }
+    sortMenus.forEach((sortMenu) => {
+      sortMenu.selectedIndex = 0;
+    });
+  });
+
+// Add event listener to the categories section of the pop up menu
+// small screens
+document
+  .querySelector(".menu-content .categories")
+  .addEventListener("click", (event) => {
+    let categoryTitle = event.target.id;
+    if (!categoryTitle) {
+      categoryTitle = event.target.parentElement.id;
+    }
+    if (categoryTitle) {
+      productsCurrentlyShown = filterByCategory(categoryTitle);
+      populateProduct(productsCurrentlyShown);
+      menuContent.classList.remove("show");
+      burgerMenu.classList.remove("open");
+    }
+    menuFilter.selectedIndex = 0;
+  }); 
+
+// Filter By Price
+// Get handle of the Filter drop downs in both big and medium screens
+const sortMenus = Array.from(document.querySelectorAll(".sort-drop-menu"));
+// Add event listener to the filter by price
+// drop down menus in both big and medium screens
+sortMenus.forEach((sortMenu) => {
+  sortMenu.addEventListener("change", (event) => {
+    const target = event.target;
+    const priceRange = target.options[target.selectedIndex].label;
+    if (priceRange === "Select Price Range") {
+      populateProduct(productsCurrentlyShown);
+    } else {
+      filterProducts(priceRange);
+    }
+  });
 });
+
+// Get handle of filter by price menu in small screens (menu content)
+// Add event listener to it (change)
+const menuFilter = document.querySelector("#menu-sort");
+menuFilter.addEventListener("change", (event) => {
+  const target = event.target;
+  const priceRange = target.options[target.selectedIndex].label;
+  if (priceRange === "Select Price Range") {
+    populateProduct(productsCurrentlyShown);
+  } else {
+    filterProducts(priceRange);
+  }
```

Comment:
Since a lot of this change handler is the same as the big and medium screen dropdown change handler above it, you could factor out a function and call it in both places (to remove the duplication):
```
const onPriceChange = (event) => {
    const target = event.target;
    const priceRange = target.options[target.selectedIndex].label;
    if (priceRange === "Select Price Range") {
      populateProduct(productsCurrentlyShown);
    } else {
      filterProducts(priceRange);
    }
};

sortMenus.forEach((sortMenu) => sortMenu.addEventListener("change", onPriceChange));
menuFilter.addEventListener("change", (event) => {
  onPriceChange(event);
  menuContent.classList.remove("show");
  burgerMenu.classList.remove("open");
});
```

Removing duplication is a good thing because then in the future, you only have one place, instead of two, where you have to make changes or fix bugs.

Reply:
Yes of course, I'll do it while refactoring 👍